common/address.go

```diff
@@ -48,6 +48,19 @@ func (a Address) InternalAddress() (InternalAddress, error) {
 	return *internal, nil
 }
 
+func (a Address) InSameSliceAs(cmp Location) bool {
+	loc := *a.Location()
+	// Figure out which location is shorter
+	shorter := loc
+	longer := cmp
+	if len(loc) > len(cmp) {
```

Comment:
Why not use `loc.InSameSliceAs()`?

Reply:
I will just drop this commit honestly, its the same computation